### ISSUE TITLE
Empty macro is treated as diverging code

### ISSUE THREAD

**Issue by jonas-schievink** [rust-analyzer/rust-analyzer#10192]
This reports no type mismatches, even though it should:

```rust
macro_rules! panik {
    () => {};
}

fn diverge() -> ! {
    panik!()
}
```

Removing the `panik!()` call reports one: `56..59: expected !, got ()`

**Comment by jonas-schievink** [comment]
This is actually correct type checking behavior, but the macro expansion should fail because it doesn't emit an expression

**Comment by ChayimFriedman2** [comment]
It does produce a diagnostic now.
